File: JAVA/APE6_10.java
package APE6;

import java.util.LinkedList;
import java.util.Scanner;

public class APE6_10 {
    public static void main(String[] args) {
        LinkedList<String> fila = new LinkedList<>();
        Scanner sc = new Scanner(System.in);
        int opcion;
        int atendidos = 0; 
 
        System.out.println("=== SISTEMA DE FILA DE ATENCION ===");
 
        do {
            System.out.println("\nPersonas en fila: " + fila.size());
            System.out.println("1. Llega una persona nueva");
            System.out.println("2. Atender siguiente persona");
            System.out.println("3. Ver quien está en la fila");
            System.out.println("4. Ver estadisticas");
            System.out.println("5. Salir");
            System.out.print("Opción: ");
            opcion = sc.nextInt();
            sc.nextLine();
 
            switch (opcion) {
                case 1:
                // Agregar una persona al final de la fila
                System.out.print("Nombre de la persona: ");
                String persona = sc.nextLine();
                fila.add(persona);
                System.out.println(persona + " se unio a la fila. " +
                                   "Su posicion: #" + fila.size());
                break;
 
                case 2:
                // Atender al primero de la fila
                if (fila.isEmpty()) {
                    System.out.println("No hay nadie en la fila.");
                } else {
                    String siguiente = fila.peek();
                    System.out.println("Atendiendo a: " + siguiente + "...");
                    fila.poll();
                    atendidos++;
                    System.out.println("Atencion completada. Personas que faltan: " + fila.size());
                }
                break;
 
                case 3:
                // Mostrar la fila completa
                if (fila.isEmpty()) {
                    System.out.println("La fila esta vacia.");
                } else {
                    System.out.println("Fila actual (de primero a ultimo):");
                    int pos = 1;
                    for (int i=0; i<fila.size(); i++) {
                        System.out.println("  #" + pos + " -> " + fila.get(i));
                        pos++;
                    }
                }
                break;
 
                case 4:
                // Mostrar estadísticas del sistema
                System.out.println("--- Estadisticas ---");
                System.out.println("Personas atendidas: " + atendidos);
                System.out.println("Personas en espera: " + fila.size());
                if (!fila.isEmpty()) {
                    System.out.println("Proxima persona: " + fila.peek());
                }
                break;

                case 5:
                System.out.println("Saliendo del programa");
                break;

                default:
                System.out.println("Opcion Invalida!");
                break;
            }
 
        } while (opcion != 5);
 
        System.out.println("\nSistema cerrado. Total atendidos hoy: " + atendidos);
        sc.close();
    }
}

File: C++/APE6_10.cpp
#include <iostream>
#include <list>
#include <string>

using namespace std;

int main() {

    list<string> fila;
    int opcion;
    int atendidos = 0;

    cout << "=== SISTEMA DE FILA DE ATENCION ===" << endl;

    do {
        cout << "\nPersonas en fila: " << fila.size() << endl;
        cout << "1. Llega una persona nueva" << endl;
        cout << "2. Atender siguiente persona" << endl;
        cout << "3. Ver quien esta en la fila" << endl;
        cout << "4. Ver estadisticas" << endl;
        cout << "5. Salir" << endl;
        cout << "Opcion: ";
        cin >> opcion;
        cin.ignore();

        switch (opcion) {

            case 1: {
                // Agregar persona
                string persona;
                cout << "Nombre de la persona: ";
                getline(cin, persona);
                fila.push_back(persona);
                cout << persona << " se unio a la fila. "
                     << "Su posicion: #" << fila.size() << endl;
                break;
            }

            case 2:
                // Atender persona
                if (fila.empty()) {
                    cout << "No hay nadie en la fila." << endl;
                } else {
                    string siguiente = fila.front();
                    cout << "Atendiendo a: " << siguiente << "..." << endl;
                    fila.pop_front();
                    atendidos++;
                    cout << "Atencion completada. Personas que faltan: "
                         << fila.size() << endl;
                }
                break;

            case 3:
                // Mostrar fila
                if (fila.empty()) {
                    cout << "La fila esta vacia." << endl;
                } else {
                    cout << "Fila actual (de primero a ultimo):" << endl;
                    int pos = 1;
                    for (const string& persona : fila) {
                        cout << "  #" << pos << " -> " << persona << endl;
                        pos++;
                    }
                }
                break;

            case 4:
                // Estadisticas
                cout << "--- Estadisticas ---" << endl;
                cout << "Personas atendidas: " << atendidos << endl;
                cout << "Personas en espera: " << fila.size() << endl;
                if (!fila.empty()) {
                    cout << "Proxima persona: " << fila.front() << endl;
                }
                break;

            case 5:
                cout << "Saliendo del programa" << endl;
                break;

            default:
                cout << "Opcion invalida!" << endl;
                break;
        }

    } while (opcion != 5);

    cout << "\nSistema cerrado. Total atendidos hoy: "
         << atendidos << endl;

    return 0;
}